pkg/edgehub/clients/wsclient/websocket_test.go

```diff
@@ -0,0 +1,405 @@
+		wantErr bool
+	}{
+		{name: "TestInit: Success in connection ",
+			fields: fields{
+				webConn:  &websocket.Conn{},
+				sendLock: sync.Mutex{},
+				config: &WebSocketConfig{
+					URL:              "ws://" + testServer.Listener.Addr().String() + "/normal",
+					CertFilePath:     "/tmp/edge.crt",
+					KeyFilePath:      "/tmp/edge.key",
+					HandshakeTimeout: 500 * time.Second,
+					WriteDeadline:    100 * time.Second,
+					ReadDeadline:     100 * time.Second,
+					ExtendHeader:     http.Header{},
+				},
+			},
+			wantErr: false,
+		},
+		{name: "TestInit: If Certificate files not loaded properly",
+			fields: fields{
+				webConn:  &websocket.Conn{},
+				sendLock: sync.Mutex{},
+				config: &WebSocketConfig{
+					URL:              "ws://" + testServer.Listener.Addr().String() + "/normal",
+					CertFilePath:     "wrong_path.crt",
+					KeyFilePath:      "wrong_path.key",
+					HandshakeTimeout: 500 * time.Second,
+					WriteDeadline:    100 * time.Second,
+					ReadDeadline:     100 * time.Second,
+					ExtendHeader:     http.Header{},
+				},
+			},
+			wantErr: true,
+		},
+		{name: "TestInit: Error in dial call returned with valid resp object ",
+			fields: fields{
+				webConn:  &websocket.Conn{},
+				sendLock: sync.Mutex{},
+				config: &WebSocketConfig{
+					URL:              "ws://" + testServer.Listener.Addr().String() + "/bad_request",
+					CertFilePath:     "/tmp/edge.crt",
+					KeyFilePath:      "/tmp/edge.key",
+					HandshakeTimeout: 500 * time.Second,
+					WriteDeadline:    100 * time.Second,
+					ReadDeadline:     100 * time.Second,
+					ExtendHeader:     http.Header{},
+				},
+			},
+			wantErr: true,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			wcc := &WebSocketClient{
+				webConn:  tt.fields.webConn,
+				sendLock: tt.fields.sendLock,
+				config:   tt.fields.config,
+			}
+			if err := wcc.Init(); (err != nil) != tt.wantErr {
+				t.Errorf("WebSocketClient.Init() error = %v, wantErr %v", err, tt.wantErr)
+			}
+		})
+	}
+}
+
+//TestUninit tests the Uninit function by trying to access the connection object
+func TestUninit(t *testing.T) {
+	type fields struct {
+		webConn  *websocket.Conn
+		sendLock sync.Mutex
+		config   *WebSocketConfig
+	}
+	tests := []struct {
+		name   string
+		fields fields
+	}{
+		{name: "TestUninit ",
+			fields: fields{
+				webConn:  &websocket.Conn{},
+				sendLock: sync.Mutex{},
+				config: &WebSocketConfig{
+					URL:              "ws://" + testServer.Listener.Addr().String() + "/normal",
+					CertFilePath:     "/tmp/edge.crt",
+					KeyFilePath:      "/tmp/edge.key",
+					HandshakeTimeout: 500 * time.Second,
+					WriteDeadline:    100 * time.Second,
+					ReadDeadline:     100 * time.Second,
+					ExtendHeader:     http.Header{},
+				},
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			wcc := &WebSocketClient{
+				webConn:  tt.fields.webConn,
+				sendLock: tt.fields.sendLock,
```

Comment:
no need to declare new structure here, can directly use tt.fields to call the function ??